Explorer/Assets/DCL/WebRequests/GenericPost/BufferedStringUploadHandler.cs

```diff
@@ -0,0 +1,295 @@
+﻿using System;
+using System.Text;
+using Unity.Collections;
+using Unity.Collections.LowLevel.Unsafe;
+using UnityEngine.Networking;
+
+namespace DCL.WebRequests
+{
+    /// <summary>
+    ///     Custom upload handler that builds JSON directly into a byte buffer
+    ///     without intermediate string allocations.
+    /// </summary>
+    public struct BufferedStringUploadHandler : IDisposable
+    {
+        // Use a native array instead of pooling a managed one to reduce the memory consumption
+        // Native Arrays are not GCed so they don't produce the corresponding pressure
+        // Additionally, it's not needed to estimate and maintain the pool to prevent bloating up
+        private NativeList<byte> buffer;
+
+        private static readonly byte[] TRUE_BYTES = Encoding.UTF8.GetBytes("true");
+        private static readonly byte[] FALSE_BYTES = Encoding.UTF8.GetBytes("false");
+        private static readonly byte[] NULL_BYTES = Encoding.UTF8.GetBytes("null");
+
+        public BufferedStringUploadHandler(int initialCapacity = 1024)
+        {
+            buffer = new NativeList<byte>(initialCapacity, Allocator.Persistent);
+        }
+
+        /// <summary>
+        ///     Sets the buffer as the upload data.
+        ///     Call this after building your JSON.
+        /// </summary>
+        public UploadHandlerRaw CreateUploadHandler() =>
+            new (buffer.AsArray().AsReadOnly());
+
+        public void Dispose() =>
+            buffer.Dispose();
+
+        /// <summary>
+        ///     Writes a single byte to the buffer.
+        /// </summary>
+        public void WriteByte(byte value) =>
+            buffer.Add(value);
+
+        public void WriteChar(char value) =>
+            buffer.Add((byte)value);
```

Comment:
Add a comment "this will break when given non-ANSI characters". And technically, you can't be writing single chars because some code points consist of two chars.